plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java

```diff
@@ -73,6 +73,10 @@ public ServletClasspathProvider() {
     }
   }
 
+  // TODO this method is called e.g. when the user clicks through the project to drill down into the libraries attached
+  // to the project and resolves the servlet and jsp jars each time. So we must ensure that if the supported versions
+  // change in the future, resolved versions of these jars remain the same for a project as long as it is associated
+  // the same runtime https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/953
```

Comment:
I tried to add more clarification, but maybe it only became more convoluted. If so, lets chat about it and I'll explain what I mean and we can find a better way to document it.